datafusion/sqllogictest/test_files/union_by_name.slt

```diff
@@ -244,13 +244,15 @@ SELECT x, y FROM t1 UNION BY NAME (SELECT y, z FROM t2 INTERSECT SELECT 2, 2 as
 3 3 NULL
 NULL 2 2
 
-query III
-SELECT x, y FROM t1 UNION ALL BY NAME (SELECT y, z FROM t2 INTERSECT SELECT 2, 2 as two FROM t1 ORDER BY 1) ORDER BY 1;
-----
-1 1 NULL
-3 3 NULL
-3 3 NULL
-NULL 2 2
+# TODO: Test regression on a new feature, with the same SanityCheckPlan failures as currently existing for this feature.
```

Comment:
This doesn't seem right to me -- this shows a regression (a query that was working is now no longer working)? 

Doesn't that mean that the fix in this PR is likely not correct?